cmd/ci-secret-generator/main_test.go

```diff
@@ -228,3 +229,47 @@ func TestValidateContexts(t *testing.T) {
 		})
 	}
 }
+func TestExecuteCommand(t *testing.T) {
+	testCases := []struct {
+		name          string
+		cmd           string
+		expected      []byte
+		expectedError error
+	}{
+		{
+			name:     "basic case",
+			cmd:      "echo basic case",
+			expected: []byte("basic case\n"),
+		},
+		{
+			name:          "error on no output",
+			cmd:           "true",
+			expectedError: fmt.Errorf("command \"true\" returned no output"),
+		},
+		{
+			name:          "error on cmd failure",
+			cmd:           "false",
+			expectedError: fmt.Errorf(" : exit status 1"),
+		},
+		{
+			name:          "error if stderr is not empty",
+			cmd:           ">&2 echo some error",
+			expectedError: fmt.Errorf("command \">&2 echo some error\" has error output"),
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			if tc.name == "error if stderr is not empty" {
+				t.Skip("Skipping a test before https://issues.redhat.com/browse/DPTP-3033 is complete:", tc.name)
+			}
+			actual, actualError := executeCommand(tc.cmd)
+			if diff := cmp.Diff(tc.expected, actual); diff != "" {
+				t.Errorf("%s: actual does not match expected, diff: %s", tc.name, diff)
+			}
+			if diff := cmp.Diff(tc.expectedError, actualError, testhelper.EquateErrorMessage); diff != "" {
+				t.Errorf("%s: actual does not match expected, diff: %s", tc.name, diff)
+			}
```

Comment:
No need to push another version just for this, but `T::Run` already adds the test case name (the first argument) to the error messages.

Reply:
OK, this may be worth an edit after all: the diff is `expected → actual`, but the message says "actual does not match expected" — the latter is what I normally expect ([e.g.](https://doc.qt.io/qt-5/qtest.html#QCOMPARE)).

Reply:
Shit. That is why I always got confused when I have to debug the diff.

https://github.com/google/go-cmp/blob/master/cmp/example_test.go

Let me fix it.

Reply:
Yes, it's moderately annoying that the division between want/got and got/want is close to 50% in these things.

I'm fine with either (I bet we have both in our code), but the message has to reflect the order, otherwise it's impossible to decode the error message, for exactly this reason.

Reply:
Updated with a clear - and + meaning as the example.